laser_slam_ros/include/laser_slam_ros/laser_slam_worker.hpp

```diff
@@ -43,6 +43,10 @@ class LaserSlamWorker {
   // Get a filtered map and apply map separation if desired.
   void getFilteredMap(laser_slam_ros::PointCloud* filtered_map);
 
+  // Get a vector containing the point clouds that have been recorded since
+  // the last call to this method. This call clears the point cloud queue.
+  std::vector<laser_slam_ros::PointCloud> getQueuedPoints();
```

Comment:
Not really... return by value here in such a case is optimized by the compiler using move semantics. Writing to the pointer would be a move as well. There is no relevant performance difference.

Since you already doing return by pointer I'd go for consistency and continue like you did. It's in some sense better because you clearly transfer the ownership of the data to the caller.
One exception: const refs to vectors (like the downsampled view of the voxel grid) can be returned by value (the ownership clearly remains to the caller).